deepmd/pt/train/training.py

```
@@ -247,6 +249,60 @@ def get_dataloader_and_iter(
                 valid_numb_batch,
             )
 
+        def compute_total_numb_batch(
+            numb_batches: Iterable[int],
+            sampler_weights: np.ndarray,
+        ) -> int:
+            weights = np.asarray(sampler_weights, dtype=np.float64)
+            if weights.ndim != 1:
+                raise ValueError("Sampler weights must be 1D.")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)